tests/scripts/all.sh

```diff
@@ -692,194 +688,191 @@ make CC=gcc CFLAGS='-Werror -Wall -Wextra'
 
-msg "build: Default + MBEDTLS_RSA_NO_CRT (ASan build)" # ~ 6 min
+msg "build: cmake 'out-of-source' build" # ~ 20s
 cleanup
-cp "$CONFIG_H" "$CONFIG_BAK"
-scripts/config.pl set MBEDTLS_RSA_NO_CRT
-CC=gcc cmake -D CMAKE_BUILD_TYPE:String=Asan .
+MBEDTLS_ROOT_DIR="$PWD"
+mkdir "$OUT_OF_SOURCE_DIR"
+cd "$OUT_OF_SOURCE_DIR"
+cmake "$MBEDTLS_ROOT_DIR"
 make
 
-msg "test: MBEDTLS_RSA_NO_CRT - main suites (inc. selftests) (ASan build)"
+msg "test: cmake 'out-of-source' build" # < 10s
 make test
+cd "$MBEDTLS_ROOT_DIR"
+rm -rf "$OUT_OF_SOURCE_DIR"
 
-msg "build: Windows cross build - mingw64, make (Link Library)" # ~ 30s
-cleanup
-make CC=i686-w64-mingw32-gcc AR=i686-w64-mingw32-ar LD=i686-w64-minggw32-ld CFLAGS='-Werror -Wall -Wextra' WINDOWS_BUILD=1 lib programs
 
-# note Make tests only builds the tests, but doesn't run them
-make CC=i686-w64-mingw32-gcc AR=i686-w64-mingw32-ar LD=i686-w64-minggw32-ld CFLAGS='-Werror' WINDOWS_BUILD=1 tests
-make WINDOWS_BUILD=1 clean
+if [ $RUN_ARMCC -ne 0 ] && [ $YOTTA -ne 0 ]; then
+    # Note - use of yotta is deprecated, and yotta also requires armcc to be on the
+    # path, and uses whatever version of armcc it finds there.
+    msg "build: create and build yotta module" # ~ 6 min 30s
+    cleanup
+    record_status tests/scripts/yotta-build.sh
+fi
+
 
-msg "build: Windows cross build - mingw64, make (DLL)" # ~ 30s
-make CC=i686-w64-mingw32-gcc AR=i686-w64-mingw32-ar LD=i686-w64-minggw32-ld CFLAGS='-Werror -Wall -Wextra' WINDOWS_BUILD=1 SHARED=1 lib programs
-make CC=i686-w64-mingw32-gcc AR=i686-w64-mingw32-ar LD=i686-w64-minggw32-ld CFLAGS='-Werror -Wall -Wextra' WINDOWS_BUILD=1 SHARED=1 tests
-make WINDOWS_BUILD=1 clean
+
+################################################################
+#### 5. Slow outside Linux-x86_64: MemSan or memcheck
+################################################################
 
 # MemSan currently only available on Linux 64 bits
 if uname -a | grep 'Linux.*x86_64' >/dev/null; then
 
-    msg "build: MSan (clang)" # ~ 1 min 20s
+    msg "build: MSan (clang)" # ~ 1 min
     cleanup
     cp "$CONFIG_H" "$CONFIG_BAK"
+    scripts/config.pl full
+    scripts/config.pl unset MBEDTLS_MEMORY_BACKTRACE # too slow for tests
     scripts/config.pl unset MBEDTLS_AESNI_C # memsan doesn't grok asm
     CC=clang cmake -D CMAKE_BUILD_TYPE:String=MemSan .
     make
 
-    msg "test: main suites (MSan)" # ~ 10s
+    msg "test: main suites (MSan)" # < 10s
     make test
 
-    msg "test: ssl-opt.sh (MSan)" # ~ 1 min
+    msg "test: ssl-opt.sh (MSan)" # ~ 3 min
     if_build_succeeded tests/ssl-opt.sh
 
     # Optional part(s)
 
     if [ "$MEMORY" -gt 0 ]; then
-        msg "test: compat.sh (MSan)" # ~ 6 min 20s
+        msg "test: compat.sh (MSan)" # ~ 6 min
         if_build_succeeded tests/compat.sh
     fi
 
 else # no MemSan
 
     msg "build: Release (clang)"
     cleanup
+    scripts/config.pl full
     CC=clang cmake -D CMAKE_BUILD_TYPE:String=Release .
```

Comment:
Why ```Release``` build is done only when ```MemSan``` is not possible. Are the equivalent alternatives? 
Should we run ```Release``` always? Please document the rationale.

Reply:
I think that's a very fair question but I'd rather leave that to a later stage (just like Gilles' point about the fact that we might want valgrind in addition to rather than instead of memsan): as soon as your work on moving the CI scripts to the main repo is done, I expect we'll have a time where we try to rationalize what's covered in the PR vs nightly/release jobs, and I expect that will be a more appropriate time to resolve that.

Currently there is not clear rationale indeed, and one of the goals of this PR is to prepare for some rationalisation work, but not do it right now.